x/gamm/types/codec.go

```diff
@@ -14,6 +14,8 @@ func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
 	cdc.RegisterInterface((*TraditionalAmmInterface)(nil), nil)
 	cdc.RegisterConcrete(&MsgJoinPool{}, "osmosis/gamm/join-pool", nil)
 	cdc.RegisterConcrete(&MsgExitPool{}, "osmosis/gamm/exit-pool", nil)
+	cdc.RegisterConcrete(&MsgSwapExactAmountIn{}, "osmosis/gamm/swap-exact-amount-in", nil)
+	cdc.RegisterConcrete(&MsgSwapExactAmountOut{}, "osmosis/gamm/swap-exact-amount-out", nil)
```

Comment:
Why are only these getting registered and not the others? (num pools, create pool, etc)

Reply:
create pool is handled per-model:
https://github.com/osmosis-labs/osmosis/blob/f0bb20932d8c3468926f4b3a4935d6ad245d3f37/x/gamm/pool-models/balancer/codec.go#L17

That was unchanged. Queries do not need this registration, only messages